cpp/CMakePresets.json

```diff
@@ -117,17 +117,46 @@
         "ARROW_GANDIVA": "ON"
       }
     },
+    {
+      "name": "features-python-minimal",
+      "inherits": [
+	  "features-minimal"
+      ],
+      "hidden": true,
+      "cacheVariables": {
+        "ARROW_COMPUTE": "ON",
+        "ARROW_CSV": "ON",
+        "ARROW_FILESYSTEM": "ON",
+        "ARROW_JSON": "ON"
```

Comment:
I haven't yet used the presets myself, but shall we include `ARROW_DATASET` here as well? (that was included with the previous `features-python` preset. 
Maybe you can also keep the exact name to keep it working for people that were using those presets?

cc @wjones127

Reply:
I think I'm fine not including datasets if it's not required for the Python build. In most workflows with CMakePresets.json, I think we expect developers to create their own presets in CMakeUserPresets.json, which will inherit from the provided once. (See [an example here](https://www.datawill.io/posts/apache-arrow-development-cpp-conda-macos/#cmakeuserpresets.json))

Once this is merged, I can send a notice to the mailing list with instructions on how to transition to the new presets. (I need to update my blog post anyways.)

Reply:
> Maybe you can also keep the exact name to keep it working for people that were using those presets?

It makes sense. I added no `-minimal`/`-maximal` versions.